dsgrid/dataset/dataset_schema_handler_one_table.py

```diff
@@ -79,3 +86,15 @@ def _check_one_table_data_consistency(self, config: DatasetConfig, load_data):
                 raise DSGInvalidDataset(
                     f"load_data records do not match dimension records for {name}"
                 )
+        dim_records_combo = set(itertools.product(*dim_records_list))
+        data_records_combo = get_unique_values(load_data_df, dim_names)
```

Comment:
I'm wondering if it's better to implement this check in Spark instead of Python. Would it be guaranteed to work for TEMPO data if we used the one-table format for it? I'm expecting 53 million unique time arrays for it. We are configuring Spark with a max result size (output of `df.collect()`) of 4 GiB.

If we're safe, I'm fine with it.

Reply:
Here we check all dimension combinations without adding in time. Time is checked separately in `_check_dataset_time_consistency` where we make sure each of these combinations see the same number of timestamps.

Reply:
How does the math work out? Max number of time arrays * number of dimension columns * size-per-cell

Reply:
Per conversation on 02/02/2022, the combo check will not be necessary at the dataset level.